### PR TITLE
configure opening repo in github codespaces

### PR DESCRIPTION
**Pull request by david-wolgemuth**
adds a button to the readme:

Screenshot 2024-02-22 at 12 55 34

will install python/node and forward port 8000,

end result will look like:

Screenshot 2024-02-22 at 12 56 03
Screenshot 2024-02-22 at 12 56 01

(side note - not sure why i'm only seeing `site.css`, and nothing else in `out` after running `make`) ??

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/universal:2",
+  "features": {
+    "ghcr.io/devcontainers/features/node:1": {},
+    "ghcr.io/devcontainers/features/python:1": {
+      "version": "3.10.13"
```
Comment (david-wolgemuth) [inline .devcontainer/devcontainer.json:6]:
```suggestion
      "version": "3.12"
```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "codespaces": {
+      "openFiles": [
+        "README.md",
+        "Makefile",
+        "data/groups/python-frederick.md",
+        "techcity/cli.py"
+      ]
+    }
+  },
+  "postCreateCommand": "make bootstrap && make run"
```
Comment (david-wolgemuth) [inline .devcontainer/devcontainer.json:19]:
```suggestion
  "postCreateCommand": "make bootstrap"
```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/universal:2",
```
Comment (david-wolgemuth) [inline .devcontainer/devcontainer.json:2]:
```suggestion
  "image": "mcr.microsoft.com/devcontainers/universal:2",
  "forwardPorts": [8000],
```

**Comment by david-wolgemuth** [comment]
ping @mblayman - this is ready!

**Comment by mblayman** [comment]
I just fixed the default `make` target (i.e., `run`) so that it will depend on the `build` target. That ensure that running `make` will always generated the full site content.

**Pull request merged**